tests/system/test_ip_types.py

```diff
@@ -45,19 +47,21 @@ def getconn() -> pymysql.connections.Connection:
 
 
 def test_public_ip() -> None:
-    try:
-        pool = init_connection_engine(IPTypes.PUBLIC)
-    except Exception as e:
-        logging.exception("Failed to initialize pool with public IP", e)
-    with pool.connect() as conn:
-        conn.execute("SELECT 1")
+    with Connector() as connector:
+        try:
+            pool = init_connection_engine(connector, IPTypes.PUBLIC)
+        except Exception as e:
+            logging.exception("Failed to initialize pool with public IP", e)
```

Comment:
Should we allow this exception to bubble up? Catching it means getting a None exception later instead.